test/TestRoot.js

```diff
@@ -0,0 +1,110 @@
+const Root = artifacts.require('./Root.sol');
+const DNSSEC = artifacts.require('./mocks/DummyDNSSEC.sol');
+const ENS = artifacts.require('./ENSRegistry.sol');
+
+const utils = require('./helpers/Utils.js');
+const namehash = require('eth-ens-namehash');
+const dns = require('../lib/dns.js');
+
+contract('Root', function(accounts) {
+
+    let node;
+    let ens, dnssec, root;
+
+    let now = Math.round(new Date().getTime() / 1000);
+
+    beforeEach(async function() {
+        node = namehash.hash('eth');
+
+        ens = await ENS.new();
+        dnssec = await DNSSEC.new();
+        root = await Root.new(ens.address, dnssec.address);
+
+        await ens.setSubnodeOwner(0, web3.sha3('eth'), root.address, {from: accounts[0]});
+        await ens.setOwner(0, root.address);
+    });
+
+    describe('setSubnodeOwner', async () => {
+
+        it('should fail when trying to set subnode owner for non root domain', async () => {
+            try {
+                await root.setSubnodeOwner(web3.sha3('eth'), '0x123', accounts[1], {from: accounts[0]});
+            } catch (error) {
```

Comment:
Won't this silently succeed if no error occurs? See the `expectFailure` function in, eg, dnssec-oracle for an example of how to handle this easily.

Reply:
yeah I am missing an assert.fail